src/pybind/mgr/dashboard/controllers/rbd_mirroring.py

```diff
@@ -574,3 +578,42 @@ def set(self, pool_name, peer_uuid, cluster_name=None, client_id=None,
             rbd.RBD().mirror_peer_set_attributes(ioctx, peer_uuid, attributes)
 
         _reset_view_cache()
+
+
+@UIRouter('/block/mirroring', Scope.RBD_MIRRORING)
+class RbdMirroringStatus(BaseController):
+    @EndpointDoc('Display RBD Mirroring Status')
+    @Endpoint()
+    @ReadPermission
+    def status(self):
+        status = {'available': True, 'message': None}
+        orch_status = OrchClient.instance().status()
+
+        # if the orch is not available we can't create the service
+        # using dashboard.
+        if not orch_status['available']:
+            return status
+        if not CephService.get_service_list('rbd-mirror') or not CephService.get_pool_list('rbd'):
+            status['available'] = False
+            status['message'] = 'RBD mirroring is not configured'  # type: ignore
+        return status
+
+    @Endpoint('POST')
+    @EndpointDoc('Configure RBD Mirroring')
+    @CreatePermission
+    def configure(self):
+        pool = Pool()
+        service = Service()
+
+        service_spec = {
+            'service_type': 'rbd-mirror',
+            'placement': {},
+            'unmanaged': False
+        }
+
+        if not CephService.get_service_list('rbd-mirror'):
+            service.create(service_spec, 'rbd-mirror')
+
+        if not CephService.get_pool_list('rbd'):
+            pool.create('rbd-mirror', 1, 'replicated',
+                        application_metadata=['rbd'], rule_name='replicated_rule')
```

Comment:
If we want to make it visible to other devels that these are the opinionated/default settings for the RBD pool, what about moving this to the controller.pool (that should have rather been in a service) and creating a specialized class:
```py
class RBDPool(Pool):
  def create(self, name='rbd'):
      super().create(name, type='replicated', rule_name='replicated_rule', application_metadata=['rbd'])
```

Reply:
I've tried to create a service for the pool, but it was a hell of a cleanup. so didn't do it. But this is good, so I made the changes. Only concern is I got an [argument-differ](https://pylint.pycqa.org/en/latest/user_guide/messages/warning/arguments-differ.html) lint error which looks crucial. I ignored it, but not sure these should be ignored.

Reply:
From the link you shared:
- Add default arguments to the child class

That could work. Anyway, not a key thing.